src/utils.jl

```diff
@@ -123,23 +123,12 @@ end
 
 # Handling ComponentArrays
 # NOTE(@avik-pal): We should probably upsteam some of these
-function Base.zero(c::ComponentArray{T, N, <:CuArray{T}}) where {T, N}
-    return ComponentArray(zero(getdata(c)), getaxes(c))
-end
-
 Base.vec(c::ComponentArray) = getdata(c)
```

Comment:
This function can also be removed now. It will return a componentarray but if broadcasting is working then that is the right thing to do